src/test/unit/VoteWeigherBaseUnit.t.sol

```diff
@@ -0,0 +1,664 @@
+        cheats.prank(serviceManagerOwner);
+        serviceManager = IServiceManager(address(new OwnableMock()));
+
+        voteWeigherImplementation = new VoteWeigherBaseHarness(strategyManager, serviceManager);
+
+        voteWeigher = VoteWeigherBaseHarness(address(new TransparentUpgradeableProxy(address(voteWeigherImplementation), address(proxyAdmin), "")));
+    }
+
+    function testCorrectConstructionParameters() public {
+        assertEq(address(voteWeigherImplementation.strategyManager()), address(strategyManager));
+        assertEq(address(voteWeigherImplementation.slasher()), address(strategyManager.slasher()));
+        assertEq(address(voteWeigherImplementation.delegation()), address(strategyManager.delegation()));
+        assertEq(address(voteWeigherImplementation.serviceManager()), address(serviceManager));
+    }
+
+    function testCreateQuorum_Valid(IVoteWeigher.StrategyAndWeightingMultiplier[] memory strategiesAndWeightingMultipliers) public {
+        strategiesAndWeightingMultipliers = _convertToValidStrategiesAndWeightingMultipliers(strategiesAndWeightingMultipliers);
+        // create a quorum from the serviceManagerOwner
+        // get the quorum count before the quorum is created
+        uint8 quorumCountBefore = voteWeigher.quorumCount();
+        cheats.prank(serviceManagerOwner);
+        // expect each strategy to be added to the quorum
+        for (uint i = 0; i < strategiesAndWeightingMultipliers.length; i++) {
+            cheats.expectEmit(true, true, true, true, address(voteWeigher));
+            emit StrategyAddedToQuorum(quorumCountBefore, strategiesAndWeightingMultipliers[i].strategy, strategiesAndWeightingMultipliers[i].multiplier);
+        }
+        // created quorum will have quorum number of the count before it was created
+        cheats.expectEmit(true, true, true, true, address(voteWeigher));
+        emit QuorumCreated(quorumCountBefore);
+        voteWeigher.createQuorum(strategiesAndWeightingMultipliers);
+        
+        assertEq(voteWeigher.quorumCount(), quorumCountBefore + 1);
+        // check that all of the weights are correct
+        for (uint i = 0; i < strategiesAndWeightingMultipliers.length; i++) {
+            IVoteWeigher.StrategyAndWeightingMultiplier memory strategyAndWeightingMultiplier = voteWeigher.strategyAndWeightingMultiplierForQuorumByIndex(quorumCountBefore, i);
+            assertEq(address(strategyAndWeightingMultiplier.strategy), address(strategiesAndWeightingMultipliers[i].strategy));
+            assertEq(strategyAndWeightingMultiplier.multiplier, strategiesAndWeightingMultipliers[i].multiplier);
+        }
+    }
+
+    function testCreateQuorum_FromNotServiceManagerOwner_Reverts(
+        address notServiceManagerOwner,
+        IVoteWeigher.StrategyAndWeightingMultiplier[] memory strategiesAndWeightingMultipliers
+    ) public {
+        cheats.assume(notServiceManagerOwner != serviceManagerOwner);
```

Comment:
you'll also need to assume the `notServiceManagerOwner` is not the `proxyAdmin` address or else you'll get flaky test failures (because . Our other tests define a mapping of addresses to exclude from fuzzed addresses and then define a `fuzzedAddress` or similarly-named modifier, to reuse for this kind of filtering.

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/53/commits/d4be2b7f8585ad8ef9171fba9171db253b572e0b